CHANGELOG.md

```diff
@@ -1,3 +1,43 @@
+## 3.9.0 
+
+NEW RESOURCES:
+
+* 0519c96 - Add gitlab_repository_file resource (#724)
+* a915ccb - Add the project access token (#588)
+* 8564a07 - Implement support for the Custom Attribute APIs (#731)
+* 39b0b6d - Add support for gitlab group badges (#673)
+* dd0da2b - Implement configuration of the integration "Microsoft Teams" (#308) (#784)
+
+NEW DATA SOURCES:
+
+* febe745 - Add data sources gitlab_project_protected_branch(es) (#551)
+
+FEATURES:
+
+* b4d4f8d - Support `early_auth_check` flag in provider config (#787)
+* 500361c - Add scripts to locally lint, like ci does (#793)
+
+ENHANCEMENTS:
+
+* 9863a61 - Auto Generate Docs (#802) 
+* 84d729e - Improve documentation around authentication with GitLab (#803)
+* a9edc4a - Add environment scope to gitlab_group_variable (#717)
+* c5a2f34 - Support issues_template and merge_requests_template in project resource (#796)
+* 18f9815 - Add CONTRIBUTING.md (#786)
+* 65b8e9c - Add minimal access level permissions (#747)
+* 79a17cb - Upgrade to SDK v2 (#725)
```

Comment:
I think this needs to be called out separately in a NOTES section, since it impacts the minimum required Terraform version.

```suggestion
```